src/Features/Core/Portable/ValueTracking/ValueTracker.FindReferencesProgress.cs

```diff
@@ -172,22 +173,43 @@ private async Task TrackMethodInvocationArgumentsAsync(ReferenceLocation referen
             var originalNode = syntaxRoot.FindNode(span);
 
             if (originalNode is null)
-            {
                 return;
-            }
 
             var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
             var invocationSyntax = originalNode.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsInvocationExpression);
             if (invocationSyntax is null)
-            {
                 return;
-            }
 
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var operation = semanticModel.GetOperation(invocationSyntax, cancellationToken);
             if (operation is not IInvocationOperation)
-            {
                 return;
+
+            if (_parameterSymbol is not null)
+            {
+                // Filter out invocations not containing am argument matching _parameterSymbol
+                var containsMatchingArg = false;
+                var invocationDescendants = invocationSyntax.DescendantNodes();
+                foreach (var descendantNode in invocationDescendants)
+                {
+                    if (syntaxFacts.IsArgument(descendantNode))
+                    {
+                        // Ensure this is an argument to the requested invocation
+                        if (invocationSyntax != descendantNode.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsInvocationExpression))
+                            continue;
+
+                        var argumentOperation = semanticModel.GetOperation(descendantNode, cancellationToken);
+                        if (argumentOperation is IArgumentOperation { Parameter: { } argumentParameter } &&
+                            _parameterSymbol.Locations.SequenceEqual(argumentParameter.Locations, arg: default(VoidResult), (loc1, loc2, _) => loc1 == loc2))
```

Comment:
>_parameterSymbol.Locations.SequenceEqual(argumentParameter.Locations, arg: default(VoidResult), (loc1, loc2, _) => loc1 == loc2) [](http://example.com/codeflow?start=28&length=128)

There has to be a better way to verify the symbols are the same, but the symbol equality checks were all failing.

Reply:
Hrm, maybe I did something wrong before, I just tried it and it worked that time. Let me test this a bit more.

Reply:
Yep, seems to work now. Good, I guess.